docs/source/nlp/text_classification.rst

```diff
@@ -1,12 +1,169 @@
+
+.. code::
+
+    saw how bad this movie was  0
+    lend some dignity to a dumb story   0
+    the greatest musicians  1
+
+You may need separate files for train, validation and test with this format.
+
+Dataset Conversion
+------------------
+
+If your dataset is stored in another format, you need to convert it to NeMo's format to use this model.
+There are some conversion scripts available for datasets: SST2, IMDB, ChemProt, and THUCnews. They can to convert them from their original format to NeMo's format.
+To convert the original datasets to NeMo's format, you can use 'examples/text_classification/data/import_datasets.py' script as the following example:
+
+.. code::
+    python import_datasets.py \
+        --dataset_name DATASET_NAME \
+        --target_data_dir TARGET_PATH \
+        --source_data_dir SOURCE_PATH
+
+It reads the dataset specified by DATASET_NAME from SOURCE_PATH and converts it to NeMo's format. Then saves the new dataset at TARGET_PATH.
+
+Arguments:
+
+- dataset_name: name of the dataset to convert ("sst-2", "chemprot", "imdb", and "thucnews" are currently supported)
+- source_data_dir: directory of your dataset
+- target_data_dir: directory to save the converted dataset
+
+After the conversion, the TARGET_PATH should contain the following files:
+
+.. code::
+
+   .
+   |--TARGET_PATH
+     |-- train.tsv
+     |-- dev.tsv
+     |-- test.tsv
+
+Some datasets do not have the test set or their test set does not have any labels, therefore the corresponding file may be missing.
+
+Model Training
+--------------
+You may find an example of a config file to be used for training of the text classification model at `NeMo/examples/nlp/text_classification/conf/text_classification_config.yaml <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/text_classification/conf/text_classification_config.yaml>`__.
+You can change any of these parameters directly from the config file or update them with the command line arguments.
+
+The config file contains three main sections:
+- trainer: Trainer section contains the configs for PTL training and you may find more info at
+:doc:`../../introduction/core.html#model-training` and
+`PTL Trainer class API <https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer-class-api>'.
+- exp_manager: the configs of experiment manager. You can find more info at :doc:`../../introduction/core.html#experiment-manager`
+- model: contains the configs of the datasets, model architecture, tokenizer, optimizer, scheduler, etc.
+
+Some sub-sections of the model section are shared among most of the NLP models. The details of these sections can be found at:
+- tokenizer and language_model: The configs of the tokenizer and encoder part and the :doc:`../nlp_model.html#important-parameters`
+- optim: the configs of the optimizer and scheduler :doc:`../../introduction/core.html`
```

Comment:
Really nice how you added also these links, do you mind adding this section to the NLPModel.rst, so the other models can reuse it?

Reply:
You mean the links to the optim, tokenizer, and language model?

Reply:
I meant links to the core/trainer sections. Some other models are referring to NlpModel.rst, would be nice to have them all in one place and not copy to all the models.rst files.